File: src/main/java/hzt/aoc/day04/PassportProcessingPart1.java
package hzt.aoc.day04;

import hzt.aoc.day04.model.Passport;
import hzt.collections.ListX;

public class PassportProcessingPart1 extends Day04Challenge {

    public PassportProcessingPart1() {
        super("part 1", "Find the number of valid passports. ");
    }

    @Override
    protected long calculateResult(ListX<Passport> passports) {
        return passports.count(Passport::requiredFieldsPresent);
    }

}

File: src/main/java/hzt/aoc/day04/PassportProcessingPart1.kt
package hzt.aoc.day04

import hzt.aoc.day04.model.Passport

class PassportProcessingPart1 : Day04Challenge("part 1", "Find the number of valid passports. ") {
    override fun calculateResult(passports: List<Passport>): Long {
        return passports.stream().filter { obj: Passport -> obj.requiredFieldsPresent() }.count()
    }
}
